### PR TITLE
feat(ext/fs): stabilize `Deno.FsFile.syncData[Sync]()` and `Deno.FsFile.sync[Sync]()`

### LINKED ISSUE #22230: suggestion: finalize replacements for `Deno.fdatasync[Sync]()` and `Deno.fsync[Sync]()`
The following I/O APIs should be replaced with `Deno.FsFile` methods. This has partially been done already. However, we haven't been able to finalize the new API names, as indicated in #22085.
1. `Deno.fdatasync[Sync]()` (currently `Deno.FsFile.syncData[Sync]()`)
2. `Deno.fsync[Sync]()` (currently `Deno.FsFile.sync[Sync]`)

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #22230

### CODE REVIEW

**Review by lucacasonato — APPROVED** [review]
LGTM!

Let's wait until @bartlomieju's approval before merging though :)

**Review by bartlomieju — APPROVED** [review]
LGTM, please wait until v1.44 to merge it.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**
